sycl/include/syclcompat/util.hpp

```diff
@@ -919,6 +920,42 @@ class group : public group_base<dimensions> {
 };
 } // namespace experimental
 
+// Calculate the number of work-groups per compute unit
+// \param [in] kernel SYCL kernel to calculate for
+// \param [in] q SYCL queue used to execute kernel
+// \param [in] wg_dim3 dim3 representing work-group shape
+// \param [in] local_mem_size Local memory usage per work-group in bytes
+// \return size_t representing maximum work-groups per compute unit
+template <class KernelName>
+size_t max_active_work_groups_per_cu(KernelName kernel, sycl::queue q,
```

Comment:
minor nit: in case we go with this API:, to change it to `const KernelName& kernel`, to make it universally binding, so that the copy constructor need not be called, and can be moved if applicable. Similarly for the other function as well.

Reply:
Went with the other API :+1: